### PR TITLE
bower support of bootstrap3 daterangepicker

### PR DESCRIPTION
**Pull request by biyeun**
@dannyroberts @esoergel @NoahCarnahan @benrudolph 

(will update affected report builder and exports ui branches accordingly)

### CODE REVIEW

File: corehq/apps/style/static/style/js/daterangepicker.config.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+$(function () {
+    'use strict';
+    $.fn.getDateRangeSeparator = function () {
+        return ' to ';
+    };
+    $.fn.createBootstrap3DateRangePicker = function(range_labels, separator) {
+        var now = moment();
+        var ranges = {};
+        ranges[range_labels['last_7_days']] = [
```
Comment (dimagimon) [inline corehq/apps/style/static/style/js/daterangepicker.config.js]: ['last_7_days'] is better written in dot notation.

File: corehq/apps/style/static/style/js/daterangepicker.config.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            moment().subtract('7', 'days').startOf('days')
+        ];
+
+        ranges[range_labels['last_month']] = [
```
Comment (dimagimon) [inline corehq/apps/style/static/style/js/daterangepicker.config.js]: ['last_month'] is better written in dot notation.

File: corehq/apps/style/static/style/js/daterangepicker.config.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            moment().subtract('1', 'months').startOf('month'),
+            moment().subtract('1', 'months').endOf('month')
+        ];
+
+        ranges[range_labels['last_30_days']] = [
```
Comment (dimagimon) [inline corehq/apps/style/static/style/js/daterangepicker.config.js]: ['last_30_days'] is better written in dot notation.

File: corehq/apps/style/static/style/js/daterangepicker.config.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    };
+    $.fn.createBootstrap3DefaultDateRangePicker = function () {
+        this.createBootstrap3DateRangePicker(
+            {
+                'last_7_days': 'Last 7 Days',
+                'last_month': 'Last Month',
+                'last_30_days': 'Last 30 Days'
+            },
+            this.getDateRangeSeparator()
+        )
```
Comment (dimagimon) [inline corehq/apps/style/static/style/js/daterangepicker.config.js]: Missing semicolon.

File: corehq/apps/style/static/style/js/daterangepicker.config.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    };
+    $.fn.createBootstrap3DefaultDateRangePicker = function () {
+        this.createBootstrap3DateRangePicker(
+            {
+                last_7_days: 'Last 7 Days',
+                last_month: 'Last Month',
+                last_30_days: 'Last 30 Days'
+            },
+            this.getDateRangeSeparator()
+        )
```
Comment (dimagimon) [inline corehq/apps/style/static/style/js/daterangepicker.config.js]: Missing semicolon.

Comment (biyeun) [inline corehq/apps/style/static/style/js/daterangepicker.config.js]: bitch.

Comment (biyeun) [inline corehq/apps/style/static/style/js/daterangepicker.config.js]: ...but i still love you.

File: corehq/apps/style/templates/style/bootstrap3/base.html [hunk 6]
```diff
@@ -297,6 +306,14 @@
         {% endcompress %}
         {% endif %}
 
+        {% if request.use_daterangepicker %}
+        {% compress js %}
+        <script type="text/javascript" src="{% new_static 'moment/moment.js' %}"></script>
```
Comment (benrudolph) [inline corehq/apps/style/templates/style/bootstrap3/base.html]: :heart: momentjs

File: corehq/apps/style/static/style/js/daterangepicker.config.js (hunk 5 above)
Comment (biyeun) [inline corehq/apps/style/static/style/js/daterangepicker.config.js]: i think.

**Comment by benrudolph** [comment]
lgtm once tests pass

File: corehq/apps/style/templates/style/bootstrap3/base.html (hunk 6 above)
Comment (benrudolph) [inline corehq/apps/style/templates/style/bootstrap3/base.html]: though i don't see this in the bower config, where is it being imported from?

**Comment by esoergel** [comment]
(those builds didn't actually fail - I just killed them)

**Comment by benrudolph** [comment]
builds are actually somewhat useful for this. we have a compression test that makes sure all the javascript files listed in the compress block can be compressed (i.e. that file exists). in this case it's relevant

**Comment by esoergel** [comment]
@benrudolph There were 4 pending builds, so I killed 3 of them, leaving the latest running.

**Comment by benrudolph** [comment]
ah gotcha, makes sense

File: corehq/apps/style/templates/style/bootstrap3/base.html [hunk 7]
```diff
@@ -78,6 +87,16 @@
         <script type="text/javascript" src="{% new_static 'style/js/hq_extensions.jquery.js' %}"></script>
         {% endcompress %}
 
+        {# Up here because if daterangepicker is called from within a form widget, #}
+        {# the javascript requiring the config file is run before js-inline #}
+        {% if request.use_daterangepicker %}
+        {% compress js %}
+        <script type="text/javascript" src="{% new_static 'moment/moment.js' %}"></script>
```
Comment (biyeun) [inline corehq/apps/style/templates/style/bootstrap3/base.html]: @benrudolph moment.js is a dependency for daterangepicker

Comment (benrudolph) [inline corehq/apps/style/templates/style/bootstrap3/base.html]: ah i see

**Pull request merged**